aisdc/postprocessing/generate_report.py

```diff
@@ -0,0 +1,283 @@
+import json
+import os
+import pprint
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+class AnalysisModule:
+    """
+    Wrapper module for metrics analysis modules
+    """
+
+    def process_dict(self):
+        """
+        Function that produces a risk summary output based on analysis in this module
+        """
+        raise NotImplementedError()
+
+    def __str__(self):
+        raise NotImplementedError()
+
+
+class FinalRecommendationModule(AnalysisModule):
+    """
+    Module that generates the first layer of a recommendation report
+    """
+
+    def __init__(self, report: dict):
+        self.P_VAL_THRESH = 0.1
+        self.MEAN_AUC_THRESH = 0.65
+
+        self.SVM_WEIGHTING_SCORE = 5
+        self.MIN_SAMPLES_LEAF_SCORE = 3
+        self.STATISTICALLY_SIGNIFICANT_SCORE = 2
+        self.MEAN_AUC_SCORE = 4
+
+        self.report = report
+
+        self.scores = []
+        self.reasons = []
+
+    def _is_svm(self, svm_weighing_score):
+        if "model" in self.report:
+            if self.report["model"] == "SVC":
+                self.scores.append(svm_weighing_score)
+                self.reasons.append("Model is SVM (supercedes all other tests)")
+                return True
+        return False
+
+    def _rf_min_samples_leaf(self, min_samples_leaf_score):
+        if "model_params" in self.report:
+            if "min_samples_leaf" in self.report["model_params"]:
+                min_samples_leaf = self.report["model_params"]["min_samples_leaf"]
+                if min_samples_leaf < 5:
+                    self.scores.append(min_samples_leaf_score)
+                    self.reasons.append("Min samples per leaf < 5")
+
+    def _statistically_significant_auc(
```

Comment:
worth just calling the other module and then parsing it's output?
e.g.
make an instance of the statistically significant AUC module here and the use its output rather than re-computing?